prog/net-plugin/launch.sh

```diff
@@ -16,42 +16,38 @@ echo "Starting launch.sh"
 # Check if the IP range overlaps anything existing on the host
 /usr/bin/weaveutil netcheck $IPALLOC_RANGE weave
 
-STATUS=0
-/usr/bin/weaveutil is-swarm-manager 2>/dev/null || STATUS=$?
-if [ $STATUS -eq 0 ]; then
-    IS_SWARM_MANAGER=1
-elif [ $STATUS -eq 20 ]; then
-    echo "Host swarm is not \"active\"; exiting." >&2
-    exit 1
+# We need to get a list of Swarm nodes which might run the net-plugin:
+# - In the case of missing restart.sentinel, we assume that net-plugin has started
+#   for the first time via the docker-plugin cmd. So it's safe to request docker.sock.
+# - If restart.sentinel present, let weaver restore from it as docker.sock is not
```

Comment:
Does this mean we will hang on to a peer in the list which has actually been removed from the Docker Swarm?
(this could be filed as a separate issue; doesn't have to be fixed here.)